api/resource_groups_machine_test.go

```diff
@@ -95,19 +94,17 @@ func TestResourceGroupsMachineUpdate(t *testing.T) {
 
 	resourceGroup := api.NewResourceGroup("group_name",
 		api.MachineResourceGroup,
-		api.MachineResourceGroupData{
-			Props: api.MachineResourceGroupProps{
-				Description: "Updated",
-				MachineTags: []map[string]string{{"tag": "machineTag"}},
-			},
+		api.MachineResourceGroupProps{
+			Description: "Updated",
+			MachineTags: []map[string]string{{"tag": "machineTag"}},
 		},
 	)
 	assert.Equal(t, "group_name", resourceGroup.Name, "Machine Resource Group name mismatch")
 	assert.Equal(t, "MACHINE", resourceGroup.Type, "a new Machine Resource Group should match its type")
 	assert.Equal(t, 1, resourceGroup.Enabled, "a new Machine Resource Group should be enabled")
 	resourceGroup.ResourceGuid = resourceGUID
 
-	response, err := c.V2.ResourceGroups.UpdateMachineResourceGroup(resourceGroup)
+	response, err := c.V2.ResourceGroups.UpdateMachineResourceGroup(&resourceGroup)
```

Comment:
do you need to pass the pointer to this function?

Reply:
Part of the workaround for invalid request when resourceGUID is included in the request is to strip the resourceGUID prior to sending the update. We can come up with a better workaround

```
func (group *ResourceGroupData) resetResourceGUID() {
	group.ResourceGuid = ""
}
```